src/homebridge-roku.js

```diff
@@ -47,35 +66,93 @@ class RokuAccessory {
     accessoryInfo
       .setCharacteristic(Characteristic.Manufacturer, this.info.vendorName)
       .setCharacteristic(Characteristic.Model, this.info.modelName)
-      .setCharacteristic(Characteristic.Name, this.info.userDeviceName)
       .setCharacteristic(Characteristic.SerialNumber, this.info.serialNumber);
 
+    this.services.push(accessoryInfo);
     return accessoryInfo;
   }
 
-  setupSwitch() {
-    const switch_ = new Service.Switch(`${this.name} Power`, 'Power');
+  setupTelevision() {
+    const television = new Service.Television(this.name);
 
-    switch_
-      .getCharacteristic(Characteristic.On)
-      .on('get', callback => callback(null, this.poweredOn))
-      .on('set', (value, callback) => {
-        this.poweredOn = value;
+    television
+      .getCharacteristic(Characteristic.Active)
+      .on('get', (callback) => {
+        this.roku
+          .info()
+          .then((info) => {
```

Comment:
Can you use then/catch for all promises, instead of two argument then? I prefer the more explicit style, and this form won't catch anything thrown from the then method, which could cause an unhandled promise rejection. Some promise libraries even [consider this an anti pattern](https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-thensuccess-fail-anti-pattern).

Reply:
These were whining about Node DEP0013. I don't know much ES6 nor Node so I keysmashed until it worked. I'll try the other form.